core/store/src/trie/state_snapshot.rs

```diff
@@ -128,24 +128,21 @@ impl StateSnapshot {
 }
 
 /// Information needed to make a state snapshot.
-#[derive(Debug, PartialEq, Eq)]
-pub enum StateSnapshotConfig {
-    /// Don't make any state snapshots.
-    Disabled,
-    Enabled {
-        home_dir: PathBuf,
-        hot_store_path: PathBuf,
-        state_snapshot_subdir: PathBuf,
-        compaction_enabled: bool,
-    },
+#[derive(Default)]
+pub struct StateSnapshotConfig {
```

Comment:
This change introduces fields that are only used if `enabled = true`.
I find the original approach cleaner.

Maybe Rust expert @nagisa has an opinion on this.

Reply:
Right, I’m not sure what the motivation was behind moving away from an enum. Not only does it result in unused fields, it also eschews compile-time protections from using the configuration fields (with potentially bogus data) when the snapshots are disabled. If matching on an enum variant to obtain individual fields is the pain that’s being solved here, then something like:

```
struct StateSnapshotConfig {
    pub home_dir: PathBuf,
    pub hot_store_path: PathBuf,
    pub state_snapshot_subdir: PathBuf,
    pub compaction_enabled: bool,
}

enum StateSnapshotState {
    Disabled,
    Enabled(StateSnapshotConfig),
}

let Enabled(snapshot_config) = snapshot_state else {
    return Err(SnapshotError::SnapshotConfigDisabled);
};
```

and various similar variations can be quite good at reducing the boilerplate.

Reply:
Yes, apologies for not clarifying earlier, we have a follow up PR to use the configs even when enable is false.
https://github.com/near/nearcore/pull/9681